data_structures/src/data_request.rs

```diff
@@ -384,31 +403,52 @@ where
             .into());
         }
 
-        let mut outputs: Vec<ValueTransferOutput> = revealers
-            .iter()
-            .zip(liars.iter())
-            .filter_map(|(&revealer, &liar)| {
-                // Only reward reveals in consensus
-                if liar {
-                    None
-                } else {
-                    let vt_output = ValueTransferOutput {
-                        pkh: revealer,
-                        value: reveal_reward,
-                        time_lock: 0,
-                    };
-                    slashed_witnesses.remove(&revealer);
-                    Some(vt_output)
-                }
-            })
-            .collect();
-
-        let honests_count = outputs.len();
-        // Create tally change for the data request creator
-        if dr_output.witnesses as usize > honests_count {
-            debug!("Created tally change for the data request creator");
-            let tally_change =
-                calculate_tally_change(commits_count, reveals_count, honests_count, &dr_output);
+        let liars_count = liars.iter().fold(0, |count, liar| match liar {
+            true => count + 1,
+            false => count,
+        });
+        let slashed_count = commits_count - reveals_count + liars_count;
+        let honests_count = reveals_count - liars_count;
+        // Collateral division rest goes for the miner
```

Comment:
This is not implemented yet

Reply:
It is, in validate_tally_transaction, collateral rest is included in the tally fee

Reply:
Oh, I looked at mining.rs and the fee is not counted here